http-client/src/client.rs

```diff
@@ -79,4 +79,62 @@ impl Client for HttpClient {
 		}?;
 		jsonrpc::from_value(json_value).map_err(Error::ParseError)
 	}
+
+	async fn batch_request<T, M, P>(&self, batch: Vec<(M, P)>) -> Result<Vec<T>, Error>
```

Comment:
I think it would be possible to have:

```rust
	async fn batch_request<T, M, P>(&self, batch: &[(M, P)]) -> Result<Vec<T>, Error>
	where
		T: DeserializeOwned + Default + Clone,
		M: Into<String> + Send,
		P: Into<Params> + Send + Sync,
		for<'a> &'a M: Into<String> + Send,
		for<'a> &'a P: Into<Params> + Send;
```

It becomes super complicated.

Reply:
Fair enough. Let's use a `Vec` then, keep it simple and opinionated.